AIPrompt docs (#1883)
telerik/blazor-docs: components/aiprompt/views/commands.md

---
title: Commands View

> page_title: AIPrompt Commands View

Reviewer: ```suggestion
page_title: AIPrompt - Commands View
```

> description: Explore the AIPrompt Commands View. Learn how to define custom commands.

Reviewer: ```suggestion
description: Explore the AIPrompt's  Commands view that displays a set of predefined commands and learn how to define your custom commands.
```

slug: aiprompt-views-commands
tags: telerik,blazor,aiprompt,ai,prompt,commands
published: True
position: 30
---

# AIPrompt Commands View

> The Commands View displays a set of predefined commands the user can browse and optionally execute. The commands are passed to the component through the `Commands` parameter, which expects a collection of type `List<AIPromptCommandDescriptor>`. Commands can also be organized through a hierarchy, through a parent-child relationship.

Reviewer 2: ```suggestion
The Commands View displays a set of predefined commands, which the user can browse and execute. The commands are passed to the component through the `Commands` parameter, which expects a collection of type `List<AIPromptCommandDescriptor>`. You can also organize commands in a hierarchy, through parent-child relationships.
```

> >note Commands can be organized through a hierarchy, with a parent-child relationship. Parent commands cannot be directly executed, and only one level of nesting is supported.

Reviewer 2: This repeats the previous paragraph.

Author: I will leave only the last sentence, as this is really the important thing to note here.

## AIPromptCommandDescriptor Parameters

The following properties enable you to customize each command:

@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)

| Parameter | Type | Description |
| ----------- | ----------- | ----------- |

> | `Id` | `string` | Defines the Id of the command.

Reviewer 2: We normally skip "Defines" or similar phrases, except for boolean paramters, where it's hard to achieve it.
```suggestion
| `Id` | `string` | The Id of the command.
```

> | `Title` | `string` | The title of the command. Rendered as text within the Command View. |

Reviewer: ```suggestion
| `Title` | `string` | The title of the command. Rendered as text within the Command view. |
```

> | `Icon` | `object` | The [Telerik Font or SVG icon]({%slug common-features-icons%}) rendered before the title within the Command View. |

Reviewer: ```suggestion
| `Icon` | `object` | The [Telerik Font or SVG icon]({%slug common-features-icons%}) rendered before the title within the Command view. |
```

        },
        new AIPromptCommandDescriptor() { Id = "8", Title = "Change Formality", Icon = SvgIcon.ApplyFormat,
            Children = new List<AIPromptCommandDescriptor>
            {
                new AIPromptCommandDescriptor() { Id = "9", Title = "Casual" },
                new AIPromptCommandDescriptor() { Id = "10", Title = "Neutral" },
                new AIPromptCommandDescriptor() { Id = "11", Title = "Formal" },
            }
        },
        new AIPromptCommandDescriptor() { Id = "12", Title = "Translate", Icon = SvgIcon.EditTools,
            Children = new List<AIPromptCommandDescriptor>
            {
                new AIPromptCommandDescriptor() { Id = "13", Title = "English" },
                new AIPromptCommandDescriptor() { Id = "14", Title = "Bulgarian" },
                new AIPromptCommandDescriptor() { Id = "15", Title = "Spanish" },
            }
        },
        new AIPromptCommandDescriptor() { Id = "16", Title = "Simplify", Icon = SvgIcon.MinWidth },
        new AIPromptCommandDescriptor() { Id = "17", Title = "Expand", Icon = SvgIcon.MaxWidth },
    };

    private void HandlePromptRequest(AIPromptPromptRequestEventArgs args)
    {

>         // dummy data intentionally used. Replace the hard-coded string with a call to your AI API.

Reviewer: ```suggestion
        // This example uses dummy data intentionally. Replace the hard-coded string with a call to your AI API.
```

        args.Output = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Vel pretium lectus quam id leo in.";
    }

    private void HandleCommandExecute(AIPromptCommandExecuteEventArgs args)
    {

>         // dummy data intentionally used. Replace the hard-coded string with a call to your AI API.

Reviewer: ```suggestion
        // This example uses dummy data intentionally. Replace the hard-coded string with a call to your AI API.
```
